core-backend/sandbox/src/funcs.rs

```diff
@@ -396,8 +396,35 @@ impl<E: Ext + 'static> FuncsHandler<E> {
             .map_err(|_| {
                 ctx.trap = Some("Trapping: unable to send reply message");
                 HostError
-            });
-        result
+            })
+    }
+
+    pub fn reply_wgas(ctx: &mut Runtime<E>, args: &[Value]) -> SyscallOutput {
+        let mut args = args.iter();
+
+        let payload_ptr = pop_i32(&mut args)?;
+        let payload_len = pop_i32(&mut args)?;
+        let value_ptr = pop_i32(&mut args)?;
+        let gas_limit_ptr = pop_i32(&mut args)?;
+
+        ctx.ext
+            .with(|ext| {
+                let payload = funcs::get_vec(&ctx.memory, payload_ptr, payload_len)?;
+                let value = funcs::get_u128(&ctx.memory, value_ptr)?;
+                let gas_limit = funcs::get_u128(&ctx.memory, gas_limit_ptr)?;
+                ext.reply(ReplyPacket::new_with_gas(
+                    payload,
+                    gas_limit as GasLimit,
+                    value,
+                ))
+                .map(|_| ())
```

Comment:
Could you please also apply changes related to `on_success_code` for all replies type?